pkg/graveler/retention/garbage_collection_manager.go

```diff
@@ -145,7 +145,7 @@ func (m *GarbageCollectionManager) SaveGarbageCollectionCommits(ctx context.Cont
 		refManager:   m.refManager,
 		repositoryID: repositoryID,
 	}
-	branchIterator := ref.NewBranchIterator(ctx, m.db, repositoryID, 1000, ref.WithOrderByCommitID()) //nolint: gomnd
+	branchIterator := ref.NewDBBranchIterator(ctx, m.db, repositoryID, 1000, ref.WithOrderByCommitID()) //nolint: gomnd
```

Comment:
This direct initilization of a DB implementation now makes `GarbageCollectionManager` the `DBGarbageCollectionManager`. Instead you can pass the `GarbageCollectionManager` a `branchIteratorFactory` that accepts params and returns the appropriate implementation (DB/KV) iterator.

Reply:
I currently ignored GC completely - I will address it